azurerm/resource_arm_notification_hub_namespace.go

```diff
@@ -0,0 +1,245 @@
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Notification Hub Namespace %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmNotificationHubNamespaceRead(d, meta)
+}
+
+func resourceArmNotificationHubNamespaceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).notificationNamespacesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["namespaces"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Notification Hub Namespace %q (Resource Group %q) was not found - removing from state!", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on Notification Hub Namespace %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	sku := flattenNotificationHubNamespacesSku(resp.Sku)
+	if err := d.Set("sku", sku); err != nil {
+		return fmt.Errorf("Error setting `sku`: %+v", err)
+	}
+
+	if props := resp.NamespaceProperties; props != nil {
+		d.Set("enabled", props.Enabled)
+		d.Set("namespace_type", props.NamespaceType)
+		d.Set("servicebus_endpoint", props.ServiceBusEndpoint)
+	}
+
+	return nil
+}
+
+func resourceArmNotificationHubNamespaceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).notificationNamespacesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["namespaces"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error deleting Notification Hub Namespace %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	// the future returned from the Delete method is broken 50% of the time - let's poll ourselves for now
+	// Related Bug: https://github.com/Azure/azure-sdk-for-go/issues/2254
+	log.Printf("[DEBUG] Waiting for Notification Hub Namespace %q (Resource Group %q) to be deleted", name, resourceGroup)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"200", "202"},
+		Target:  []string{"404"},
+		Refresh: notificationHubNamespaceStateRefreshFunc(ctx, client, resourceGroup, name),
+		Timeout: 10 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Notification Hub %q (Resource Group %q) to be deleted: %s", name, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, client.Client)
```

Comment:
Do we want to comment this out until the issue mentioned above is fixed? Or is it fine to wait for when the namespace is deleted?